### PR TITLE
[IMP] base_location_geonames_import: Import several countries through the same wizard.

### PR DESCRIPTION
**Pull request by manuelregidor**
Allow to import one or more countries information selecting them altogether in the same wizard.

### CODE REVIEW

File: base_location_geonames_import/__manifest__.py [hunk 1]
```diff
@@ -17,6 +18,7 @@
         "Agile Business Group,"
         "Tecnativa,"
         "AdaptiveCity,"
+        "Sygel,"
```
Comment (pedrobaeza) [inline base_location_geonames_import/__manifest__.py]: This contribution doesn't qualify for co-authorship I'm afraid.

**Comment by pedrobaeza** [comment]
Please avoid superfluous changes, like removing a related field and change variables across code. Keep the minimal changes for the new feature.

**Comment by manuelregidor** [comment]
> Please avoid superfluous changes, like removing a related field and change variables across code. Keep the minimal changes for the new feature.

Thank you for your feedback. Please, could you indicate which changes you are refering to? Thank you.

File: base_location_geonames_import/wizard/geonames_import.py [hunk 2]
```diff
@@ -25,13 +26,8 @@
 class CityZipGeonamesImport(models.TransientModel):
     _name = "city.zip.geonames.import"
     _description = "Import City Zips from Geonames"
-    _rec_name = "country_id"
 
-    country_id = fields.Many2one("res.country", "Country", required=True)
-    code_row_index = fields.Integer(
```
Comment (pedrobaeza) [inline base_location_geonames_import/wizard/geonames_import.py:31]: These changes for example

Comment (manuelregidor) [reply base_location_geonames_import/wizard/geonames_import.py:31]: Hi! Thank you for your answer. I removed the related fields code_row_index and name_row_index because country_id field has been converted into a Many2many field (it was a Many2one field before the changes I applied). That is also why _rec_name has been removed.

Comment (pedrobaeza) [reply base_location_geonames_import/wizard/geonames_import.py:31]: OK, understood. You're right.

**Comment by manuelregidor** [comment]
Hi @pedrobaeza! I have applied some changes to the implementation and created a new test to check whether the single country import system still works under the new funcionality. Please, could you let me know if the changes performed solve the issues you pointed out? Thank you.

File: base_location_geonames_import/__manifest__.py [hunk 3]
```diff
@@ -8,7 +8,7 @@
 
 {
     "name": "Base Location Geonames Import",
-    "version": "13.0.1.0.2",
+    "version": "13.0.1.0.1",
```
Comment (pedrobaeza) [inline base_location_geonames_import/__manifest__.py]: Don't change this version number

File: base_location_geonames_import/README.rst [hunk 4]
```diff
@@ -23,7 +23,7 @@ Base Location Geonames Import
     :target: https://runbot.odoo-community.org/runbot/134/13.0
     :alt: Try me on Runbot
 
-|badge1| |badge2| |badge3| |badge4| |badge5| 
+|badge1| |badge2| |badge3| |badge4| |badge5|
```
Comment (pedrobaeza) [inline base_location_geonames_import/README.rst]: Leave this file untouched or regenerate it with the tool.

File: base_location_geonames_import/tests/test_base_location_geonames_import.py [hunk 5]
```diff
@@ -166,3 +176,93 @@ def test_import_duplicated_city_name(self):
...
+        max_import = 10
+        self.wizard_2.with_context(max_import=max_import).run_import()
+        # Look if there are imported states for the country
+        state_country_2_count = self.env["res.country.state"].search_count(
+            [("country_id", "=", self.country_2.id)]
+        )
+        state_country_3_count = self.env["res.country.state"].search_count(
+            [("country_id", "=", self.country_3.id)]
+        )
+        self.assertTrue(state_country_2_count)
```
Comment (pedrobaeza) [inline base_location_geonames_import/tests/test_base_location_geonames_import.py]: Why these asserts? You have to test features of the module, not other things. Here you are asserting the previous state of the test DB, which is not desirable. And focus the test to make sure that cities in multiple imported countries are imported, not in other things.

File: base_location_geonames_import/tests/test_base_location_geonames_import.py [hunk 6]
```diff
@@ -166,3 +176,93 @@ def test_import_duplicated_city_name(self):
...
+        self.assertEqual(city_country_2_count, city_2_count2)
+        self.assertEqual(city_country_3_count, city_3_count2)
+
+        zip_count = self.env["res.city.zip"].search_count(
+            [("city_id.country_id", "in", (self.country_2.id, self.country_3.id))]
+        )
+        self.assertEqual(zip_count, 20)
+
+    def test_import_country_single_value(self):
+        max_import = 5
```
Comment (pedrobaeza) [inline base_location_geonames_import/tests/test_base_location_geonames_import.py]: No need to add this test, and more with the same invalid asserts than the previous. For checking this backward compatibility, just don't change in https://github.com/OCA/partner-contact/pull/884/files#diff-1718593fd8e26e4fec3112ad5a4657c4R120 the assigned value and everything should work the same.

File: base_location_geonames_import/wizard/geonames_import.py [hunk 7]
```diff
@@ -79,8 +76,8 @@ def select_city(self, row, country, state_id):
             "WHERE name = %s AND country_id = %s AND state_id = %s LIMIT 1",
             (self.transform_city_name(row[2], country), country.id, state_id),
         )
-        row_city = self.env.cr.fetchone()
-        return (row_city[0], row_city[1]) if row_city else (False, False)
+        row = self.env.cr.fetchone()
```
Comment (pedrobaeza) [inline base_location_geonames_import/wizard/geonames_import.py]: Don't undo this change

**Review by pedrobaeza — CHANGES_REQUESTED** [review]
Thanks for the efforts you are doing and the work for uncoupling the methods from the model. I request one last effort in the inline comments, and please also squash all your commits in one.

File: base_location_geonames_import/tests/test_base_location_geonames_import.py (hunk 5 above)
Comment (manuelregidor) [reply base_location_geonames_import/tests/test_base_location_geonames_import.py]: Sorry, I'm not sure if I've understood what you mean. country_2 and country_3 are only used at this test, which follows the steps followed by test_import_country. Should I do the test with more than two countries and focus just on the imported cities (and not the states and zips related to each country)? Thank you.

Comment (pedrobaeza) [reply base_location_geonames_import/tests/test_base_location_geonames_import.py]: OK, I see, I got confused with so many counts. No need to test each object, as checking that there are zips, it's enough to test that the import has worked in this case. Also put `limit=1` for better performance.

**Review by HaraldPanten — APPROVED** [review]
Tested in local environment and it's working fine. Functional review 👍

**Review by ValentinVinagre — APPROVED** [review]

**Comment by manuelregidor** [comment]
@pedrobaeza  is this PR ready to merge for you? Thank you.

File: base_location_geonames_import/wizard/geonames_import.py [hunk 8]
```diff
@@ -25,13 +25,10 @@
 class CityZipGeonamesImport(models.TransientModel):
     _name = "city.zip.geonames.import"
     _description = "Import City Zips from Geonames"
-    _rec_name = "country_id"
```
Comment (pedrobaeza) [inline base_location_geonames_import/wizard/geonames_import.py:28]: If the model doesn't have `name` field, it requires a  `_rec_name` of an existing field. If you don't want `country_id`, then put `country_ids`

Comment (manuelregidor) [reply base_location_geonames_import/wizard/geonames_import.py:28]: @pedrobaeza Thank you for your feedback. The change has been applied.

**Review by pedrobaeza — APPROVED** [review]
